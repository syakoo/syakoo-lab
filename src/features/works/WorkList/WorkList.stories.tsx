import type { Meta, StoryObj } from "@storybook/react";

import { WorkList } from ".";

const meta = {
  component: WorkList,
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof WorkList>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {};
