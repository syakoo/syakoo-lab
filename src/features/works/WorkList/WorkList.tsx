import { works } from "@/contents/works/works";
import { Col, Container } from "@/design-system/layout";
import { H2, FadeIn } from "@/design-system/ui";

import { WorkBlock } from "./WorkBlock";

export const WorkList: React.FC = () => {
  return (
    <Container
      as="section"
      center
      paddingBottom="400"
      paddingX="200"
      size="100"
    >
      <Col gap="200">
        <H2>Works</H2>
        <Col as="ul" gap="100">
          {works.map((work, i) => (
            <FadeIn key={work.name} as="li" delaySec={0.1 * i}>
              <WorkBlock work={work} />
            </FadeIn>
          ))}
        </Col>
      </Col>
    </Container>
  );
};
