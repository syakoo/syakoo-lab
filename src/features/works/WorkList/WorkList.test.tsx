import { testStories } from "@/test/testStories";

import * as stories from "./WorkList.stories";

describe("WorkList", () => {
  testStories(stories);
});
