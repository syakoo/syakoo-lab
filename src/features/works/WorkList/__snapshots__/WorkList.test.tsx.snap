// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkList renders Sample 1`] = `
<body>
  <div>
    <section
      class="Container_containerStyle__1b3ljla3 Container_containerStyle_size_100__1b3ljla5 Container_containerStyle_variants_size_100__1b3ljla1 Container_containerStyle_center_true__1b3ljla7 Container_containerStyle_paddingBottom_400__1b3ljlay padding_paddingBottomVariants_400__1i0pzy6q Container_containerStyle_paddingX_200__1b3ljla13 padding_paddingXVariants_200__1i0pzy6v"
    >
      <div
        class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
      >
        <h2
          class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_500__jzpfqdp Text_textStyle_variants_size_500__jzpfqda Text_textStyle_weight_bold__jzpfqdt"
        >
          Works
        </h2>
        <ul
          class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
        >
          <li
            class="FadeIn_fadeInStyle__r7lc2h1"
            style="animation-duration: 0.5s; animation-delay: 0s;"
          >
            <article
              class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
            >
              <div
                class="WorkBlock_workBlockStyle_imageWrapper__8mmmgg1"
              >
                <img
                  alt=""
                  class="WorkBlock_workBlockStyle_image__8mmmgg2"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
              <div
                class="Flex_flexItemStyle__15ivi8419 Flex_flexItemStyle_grow_1__15ivi841d"
              >
                <div
                  class="WorkBlock_workBlockStyle_body__8mmmgg0"
                >
                  <h3
                    class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_300__jzpfqdn Text_textStyle_variants_size_300__jzpfqd8 Text_textStyle_weight_bold__jzpfqdt"
                  >
                    <a
                      class="Link_linkStyle__m4v9430"
                      href="https://magic-square.firebaseapp.com"
                    >
                      MagicSquare
                    </a>
                  </h3>
                  <div>
                    <div>
                      魔法陣のタイムアタックアプリ。初めて作った Web アプリケーションです。React + TS で作っています。
                    </div>
                    <div
                      class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_25__15ivi84w Flex_flexStyle_variants_gap_25__15ivi84b Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
                    >
                      <span
                        class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                      >
                        released: 
                        2019-08-31
                        , repo:
                         
                        <a
                          class="Link_linkStyle__m4v9430 Link_linkStyle_underlined_true__m4v9432"
                          href="https://github.com/syakoo/MagicSquare"
                        >
                          syakoo/MagicSquare
                        </a>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </article>
          </li>
          <li
            class="FadeIn_fadeInStyle__r7lc2h1"
            style="animation-duration: 0.5s; animation-delay: 0.1s;"
          >
            <article
              class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
            >
              <div
                class="WorkBlock_workBlockStyle_imageWrapper__8mmmgg1"
              >
                <img
                  alt=""
                  class="WorkBlock_workBlockStyle_image__8mmmgg2"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
              <div
                class="Flex_flexItemStyle__15ivi8419 Flex_flexItemStyle_grow_1__15ivi841d"
              >
                <div
                  class="WorkBlock_workBlockStyle_body__8mmmgg0"
                >
                  <h3
                    class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_300__jzpfqdn Text_textStyle_variants_size_300__jzpfqd8 Text_textStyle_weight_bold__jzpfqdt"
                  >
                    <a
                      class="Link_linkStyle__m4v9430"
                      href="https://syakoo.github.io/real_share_app"
                    >
                      Real Sharing
                    </a>
                  </h3>
                  <div>
                    <div>
                      Web NFC を使った秘密分散共有法のシミュレーションアプリ。React + TypeScript + Redux-saga で作成しています。
                    </div>
                    <div
                      class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_25__15ivi84w Flex_flexStyle_variants_gap_25__15ivi84b Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
                    >
                      <span
                        class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                      >
                        released: 
                        2020-04-17
                        , repo:
                         
                        <a
                          class="Link_linkStyle__m4v9430 Link_linkStyle_underlined_true__m4v9432"
                          href="https://github.com/syakoo/real_share_app"
                        >
                          syakoo/real_share_app
                        </a>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </article>
          </li>
          <li
            class="FadeIn_fadeInStyle__r7lc2h1"
            style="animation-duration: 0.5s; animation-delay: 0.2s;"
          >
            <article
              class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
            >
              <div
                class="WorkBlock_workBlockStyle_imageWrapper__8mmmgg1"
              >
                <img
                  alt=""
                  class="WorkBlock_workBlockStyle_image__8mmmgg2"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
              <div
                class="Flex_flexItemStyle__15ivi8419 Flex_flexItemStyle_grow_1__15ivi841d"
              >
                <div
                  class="WorkBlock_workBlockStyle_body__8mmmgg0"
                >
                  <h3
                    class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_300__jzpfqdn Text_textStyle_variants_size_300__jzpfqd8 Text_textStyle_weight_bold__jzpfqdt"
                  >
                    <a
                      class="Link_linkStyle__m4v9430"
                      href="https://haskell-function-puzzle.vercel.app"
                    >
                      Haskell Function Puzzle
                    </a>
                  </h3>
                  <div>
                    <div>
                      <div>
                        Haskell の関数合成パズルサイト。React + TypeScript で作成しています。
                        <a
                          class="Link_linkStyle__m4v9430 Link_linkStyle_colored_true__m4v9431"
                          href="/writings/20210316"
                        >
                          紹介サイト
                        </a>
                      </div>
                    </div>
                    <div
                      class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_25__15ivi84w Flex_flexStyle_variants_gap_25__15ivi84b Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
                    >
                      <span
                        class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                      >
                        released: 
                        2021-03-16
                        , repo:
                         
                        <a
                          class="Link_linkStyle__m4v9430 Link_linkStyle_underlined_true__m4v9432"
                          href="https://github.com/syakoo/haskell-function-puzzle"
                        >
                          syakoo/haskell-function-puzzle
                        </a>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </article>
          </li>
        </ul>
      </div>
    </section>
  </div>
</body>
`;
